### ISSUE TITLE
RUF200 does not detect invalid license field

### ISSUE THREAD

**Issue by marcocaccin** [astral-sh/ruff#13726]
`RUF200` should detect invalid contents of a `pyproject.toml` file, but the `license` field does not seem to be validated correctly.

Minimal example:
```toml
[project]
name = "xyz"
version = "0.0.1"
license = "MIT License"
```
goes through, despite the value being set incorrectly (the [specs](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license) require this to be `license = {text = "MIT License"}`). This can be particularly surprising for people transitioning from `poetry`, where the syntax above was the expected one (yet in a differently named section), see for example https://github.com/rspeer/python-ftfy/issues/218

[Info] tested ruff versions: `0.4.1`, `0.6.9`; command: `ruff check --select ALL --isolated pyproject.toml`

**Comment by MichaReiser** [comment]
That makes sense. Ruff currently doesn't support any linting for pyproject.toml or ruff toml files. We should definitely support this but it's further out. I'll merge this with https://github.com/astral-sh/ruff/issues/10487 which asks for a very similar functionality
